src/rower.cpp

```diff
@@ -3,7 +3,24 @@
 #include "qdebugfixup.h"
 #include <QSettings>
 
-rower::rower() {}
+rower::rower() {
+
+#ifdef Q_OS_IOS
+#ifndef IO_UNDER_QT
+        QSettings settings;
+        bool cadence = settings.value(QZSettings::bike_cadence_sensor, QZSettings::default_bike_cadence_sensor).toBool();
+        bool ios_peloton_workaround = settings.value(QZSettings::ios_peloton_workaround, QZSettings::default_ios_peloton_workaround).toBool();
+        bool virtual_device_rower = settings.value(QZSettings::virtual_device_rower, QZSettings::default_virtual_device_rower).toBool();
+        if (ios_peloton_workaround && cadence && !virtual_device_rower) {
+
+            qDebug() << "ios_peloton_workaround activated!";
+            this->lockScreen = new lockscreen();
+            this->lockScreen->virtualbike_ios();
```

Comment:
It's a rower, but uses the virtual bike lockscreen setup. Copied from one of the rower classes.